File: packages/types/call.ts
export interface TelecomCall {
  id: string;
  orgId: string;
  clientId?: string;
  status: TelecomCallStatus;
  phoneNumber: string;
  duration?: number;
  recordingUrl?: string;
  transcript?: string;
  complianceStatus?: string | null;
  stateCode?: string | null;
  consentCaptured?: boolean;
  gracefulGoodnightTriggered?: boolean;
  createdAt: string;
  endedAt?: string;
}

export type TelecomCallStatus =
  | "pending"
  | "dialing"
  | "ringing"
  | "in_progress"
  | "completed"
  | "failed"
  | "busy"
  | "no_answer"
  | "blocked_dnc"
  | "blocked_hours"
  | "graceful_goodnight"
  | "escalated";

export interface TriggerCallRequest {
  leadId?: number;
  agentId?: number;
  phoneNumber: string;
  campaignId?: number;
}

export interface TriggerCallResponse {
  call: {
    id: number;
    vapiCallId: string;
    orgId: string;
    leadId: number | null;
    agentId: number | null;
    campaignId: number | null;
    status: string;
    phoneNumber: string;
    complianceStatus?: string | null;
    createdAt: string | null;
  };
}

File: apps/api/models/call.py
from datetime import datetime
from sqlmodel import Field
from typing import Optional

from .base import TenantModel


class Call(TenantModel, table=True):
    __tablename__ = "calls"  # type: ignore

    vapi_call_id: str = Field(default="", max_length=255, index=True)
    lead_id: Optional[int] = Field(default=None, nullable=True, foreign_key="leads.id")
    agent_id: Optional[int] = Field(
        default=None, nullable=True, foreign_key="agents.id"
    )
    campaign_id: Optional[int] = Field(default=None, nullable=True)
    status: str = Field(default="pending", max_length=50)
    duration: Optional[int] = Field(default=None, nullable=True)
    recording_url: Optional[str] = Field(default=None, max_length=500, nullable=True)
    phone_number: str = Field(default="", max_length=20, index=True)
    transcript: Optional[str] = Field(default=None, nullable=True)
    ended_at: Optional[datetime] = Field(default=None, nullable=True)
